Tests/SwiftProtobufTests/Test_AsyncMessageSequence.swift

```diff
@@ -0,0 +1,241 @@
+// Tests/SwiftProtobufTests/Test_AsyncMessageSequence.swift - 
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Tests the decoding of binary-delimited message streams, ensuring various invalid stream scenarios are
+/// handled gracefully.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import SwiftProtobuf
+
```

Comment:
This files is going to need some swift version guards so the build on older Swift versions still works.

Reply:
Done

Reply:
Do we still need those guards with the new version support @thomasvl ?